src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1380,6 +1380,14 @@ private FileSystemInfo GetFileSystemItem(string path, ref bool isContainer, bool
             bool hidden = attributes.HasFlag(FileAttributes.Hidden);
             isContainer = attributes.HasFlag(FileAttributes.Directory);
 
+            // FileInfo allows for a file path to end in a trailing slash, but the resulting object
+            // is incomplete.  A trailing slash should indicate a directory.  So if the path ends in a
+            // trailing slash and is not a directory, return null
+            if (!isContainer && path.EndsWith(Path.DirectorySeparatorChar))
+            {
+                return null;
+            }
+
```

Comment:
The regression is in line 1378. I suggest remove it.

Reply:
removed